### ISSUE TITLE
`hosts-file-enabled` cannot be set from config file

### ISSUE THREAD

**Issue by linuxgemini** [AdguardTeam/dnsproxy#413]
Hello,

`hosts-file-enabled` uses `cmd.decodableBool` declared in `internal/cmd/options.go`:

https://github.com/AdguardTeam/dnsproxy/blob/25e110a88d8ee2bc5b8f82f872ccafca9080205c/internal/cmd/options.go#L26-L43

While the option works as intended as a launch flag `--hosts-file-enabled=false`, setting it inside a YAML config (`hosts-file-enabled: false`) will not work:

```
Aug 20 16:22:47 hostname dnsproxy[738715]: dnsproxy config path: /etc/dnsproxy/config.yaml
Aug 20 16:22:47 hostname dnsproxy[738715]: parsing config file /etc/dnsproxy/config.yaml: unmarshalling file: yaml: unmarshal errors:
Aug 20 16:22:47 hostname dnsproxy[738715]:   line 84: cannot unmarshal !!bool `false` into cmd.decodableBool
```

Setting it as a string (`hosts-file-enabled: "false"`) also fails:

```
Aug 20 16:24:47 hostname dnsproxy[738747]: dnsproxy config path: /etc/dnsproxy/config.yaml
Aug 20 16:24:47 hostname dnsproxy[738747]: parsing config file /etc/dnsproxy/config.yaml: unmarshalling file: yaml: unmarshal errors:
Aug 20 16:24:47 hostname dnsproxy[738747]:   line 84: cannot unmarshal !!str `false` into cmd.decodableBool
```

I don't understand why exactly this occurs (I'm not really proficient in Go), one might omit the YAML flag (seen below) and use it as a command line argument:

https://github.com/AdguardTeam/dnsproxy/blob/25e110a88d8ee2bc5b8f82f872ccafca9080205c/internal/cmd/options.go#L233-L235

**Comment by EugeneOne1** [comment]
@linuxgemini, hello and thanks for the detailed report.  We've just committed the fix (559d2efb48de8572b34bfd2ae95a661fdf076c9c) for this issue to the master branch.  Could you please build the `dnsproxy` binary from this commit and check if it now successfully decodes the value?

**Comment by timkgh** [comment]
The default also should not be `true`. Let users opt-in rather than opt-out by default. If you run `dnsproxy` in a cloud VM that has all sorts of local things in `/etc/hosts`, dnsproxy shouldn't really pick those up if you're using it just as a generic/public DNS proxy.

**Comment by ameshkov** [comment]
Eventually, we have in mind making `dnsproxy` a drop-in replacement for dnsmasq and that's why we're trying to follow its behavior.

**Comment by linuxgemini** [comment]
> @linuxgemini, hello and thanks for the detailed report. We've just committed the fix ([559d2ef](https://github.com/AdguardTeam/dnsproxy/commit/559d2efb48de8572b34bfd2ae95a661fdf076c9c)) for this issue to the master branch. Could you please build the `dnsproxy` binary from that commit and check if it now successfully decodes the value?

Hi @EugeneOne1, sorry for the late reply; build from current master works yes.

**Comment by EugeneOne1** [comment]
@linuxgemini, good to hear!  We'll release the patch version later today.  I'll close this issue for now.
